src/utils/get-functions.js

```diff
@@ -34,13 +34,13 @@ const getWatchDirs = (functionsSrcDir, functions) => {
   return [functionsSrcDir, ...new Set(localFilesDirs)]
 }
 
-const getFunctionsAndWatchDirs = async (functionsSrcDir) => {
+const getFunctionsAndWatchDirs = async (functionsSrcDir, functionsConfig) => {
   if (!(await fileExistsAsync(functionsSrcDir))) {
     return { functions: [], watchDirs: [functionsSrcDir] }
   }
 
   // get all functions files so we know which directories to watch
-  const functions = await listFunctionsFiles(functionsSrcDir)
+  const functions = await listFunctionsFiles(functionsSrcDir, { config: functionsConfig })
```

Comment:
Since we're listing functions files to get the watch directories, we need to pass the bundler type.

One thing I noticed is that if the bundler is `esbuild` we return the bundled function files.
e.g.

```
[
  {
    srcFile: '<repo>/.netlify/functions-serve/test/test.js',
    name: 'test',
    mainFile: '<repo>/.netlify/functions-serve/test/test.js',
    runtime: 'js',
    extension: '.js'
  }
]
```

We will need `zisi` to return the original files required by the function to make this work. Also since `functionsSrcDir` is also set to `.netlify/functions-serve` by the function builder, we'll need to handle that too.